src/widget/wsearchlineedit.cpp

```diff
@@ -58,9 +58,10 @@ void WSearchLineEdit::setDebouncingTimeoutMillis(int debouncingTimeoutMillis) {
 }
 
 WSearchLineEdit::WSearchLineEdit(QWidget* pParent)
-    : QLineEdit(pParent),
-      WBaseWidget(this),
-      m_clearButton(make_parented<QToolButton>(this)) {
+        : QLineEdit(pParent),
+          WBaseWidget(this),
+          m_clearButton(make_parented<QToolButton>(this)),
+          m_innerHeight(0) {
```

Comment:
Such dummy initialization hides the root cause from the code analysis tools like valgrind and siliences them, but does not fix the root cause.
The real problem here seems to be that m_innerHeight is only set by resizeEvent, but you can read it before using updateClearButton.
Wouldn't it be better to remove this member and use  `this->height() - 2 * m_frameWidth` in updateClearButton instead.